api/src/test/java/org/apache/iceberg/expressions/TestExpressionUtil.java

```diff
@@ -63,6 +64,21 @@ public void testSanitizeIn() {
         ExpressionUtil.toSanitizedString(Expressions.in("test", 34, 345)));
   }
 
+  @Test
+  public void testSanitizeLongIn() {
+    assertEquals(
+        Expressions.in(
+            "test", "(2-digit-int)", "(3-digit-int)", "... (63 values hidden, 65 in total) ..."),
+        ExpressionUtil.sanitize(
+            Expressions.in("test", IntStream.range(37, 102).boxed().toArray())));
```

Comment:
The conversion to strings like "2-digit-int" and "3-digit-int" should handle this already when deduped. I think a better solution for the sanitizers is to dedup using a set and then do this. Only strings would need this.